.github/workflows/codeql.yml

```
@@ -2,16 +2,66 @@ name: "CodeQL"
+            - 'cmake/**'
+            - '.github/actions/**'
+            - '.github/workflows/codeql.yml'
+          csharp:
+            - '**/*.cs'
+            - 'OloEngine-ScriptCore/**'
+            - '.github/workflows/codeql.yml'
+
   analyze:
     name: Analyze
+    needs: detect-changes
+    if: ${{ github.event_name == 'workflow_dispatch' || (matrix.language == 'c-cpp' && needs.detect-changes.outputs.cpp == 'true') || (matrix.language == 'csharp' && needs.detect-changes.outputs.csharp == 'true') }}
```

actionlint (1.7.12)

[error] context "matrix" is not allowed here. available contexts are "github", "inputs", "needs", "vars". see https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability for more details

(expression)